assignment4.Rmd

```diff
@@ -114,3 +117,34 @@ biplot(pca)
 ```
 # Part III  
 ## Also in this repository is a data set and codebook from Rod Martin, Patricia Puhlik-Doris, Gwen Larsen, Jeanette Gray, Kelly Weir at the University of Western Ontario about people's sense of humor. Can you perform a PCA on this data?
+```{r}
+dt<-read.csv('humor_data.csv')
+dt1<-dt[1:32]
+library(psych)
+fa.parallel(scale(dt1), fa="pc", n.iter=100,
+show.legend=FALSE, main="Scree plot with parallel analysis")
+#according to the scree plot, we should select 5 factors
+library(corrplot)
+C<-cor(dt1)
+#visualise the correlation of 32 variables
+corrplot(C, method="square", type="full",order="hclust",addrect = 5,
+         col=colorRampPalette(c("red", "white", "blue"))(20), tl.col="black",           tl.cex=0.6, tl.srt=90)
+#according to the corrplot, we could find that there are 4 main clusters of these question, which may indicate 4 types: affiliative,selfenhancing,aggressive,selfdefeating
+dt3<-scale(dt1)
+#principal component regression(analysis)
+prin<-principal(dt3,nfactors =5 ,scores = TRUE,rotate = "varimax")
+score<-prin$scores
+C2<-cor(score)
+#visualise the correlation of PCs
+corrplot(C2, method="square", type="full",
+         col=colorRampPalette(c("red", "white", "blue"))(20), tl.col="black",           tl.cex=0.6, tl.srt=90, addCoef.col="black", addCoefasPercent = TRUE,
+         sig.level=0.50, insig = "blank")
+biplot(prin)
+#show the loadings of PCs
+print(prin$loadings)
+dt4<-cbind(dt[33:36],score)
+corrplot(cor(dt4), method="square", type="full",
+         col=colorRampPalette(c("red", "white", "blue"))(20), tl.col="black",           tl.cex=0.6, tl.srt=90, addCoef.col="black", addCoefasPercent = TRUE,
+         sig.level=0.50, insig = "blank")
+#according to the corrplot, we could conlude that 4 rotated principal components are respectively highly correlated with 4 types the author came up with.
+```
```

Comment:
Nice work using the psych package--I've never tried it before. Your analyiss here is very good and I appreciate the interpretation that you provide at the end suggesting that there PC's are similar to what the author came up with.